aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -152,6 +167,27 @@ static Tensor &_dimreduce_setup(Tensor &result, const Tensor &self,
   return result;
 }
 
+static inline Tensor &mean_out(Tensor &result, const Tensor &self, int64_t dim,
+                 bool keepdim, optional<ScalarType> dtype) {
+  at::_sum_out(result, self.toType(result.type().scalarType()), dim, keepdim);
```

Comment:
why doesn't this call at::native::sum_out?  That checks that e.g. the result dtype matches the passed in dtype.